### PR TITLE
docs: Improves the Regression Tutorial

### LINKED ISSUE #804: Improve the regression tutorial
The current [regression tutorial](https://library.safeds.com/en/stable/tutorials/regression/) is missing important steps:

* Remove some columns completely
* Scale some columns
* ... (an DSL orientieren )

It should also use headings instead of an ordered list to populate the table of contents. Ideally, it should match the upcoming "Getting started" tutorial for the DSL closely, so it's easy to compare DSL and Python syntax.

### PR DESCRIPTION
**Pull request by TellemHD**
Closes #804 

### Summary of Changes

- Improves the Regression Tutorial

### CODE REVIEW

File: docs/tutorials/regression.ipynb [hunk 1]
```diff
@@ -2,204 +2,391 @@
  "cells": [
   {
    "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
    "source": [
     "# Regression\n",
     "\n",
     "This tutorial uses safeds on **house sales data** to predict house prices.\n"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:11]:
```suggestion
    "This tutorial uses Safe-DS on **house sales data** to predict house prices.\n"
```

File: docs/tutorials/regression.ipynb [hunk 2]
```diff
@@ -2,204 +2,391 @@
...
    "metadata": {
     "collapsed": false
-   }
+   },
+   "source": [
+    "## File and Imports\n",
+    "\n",
+    "Start by creating a Python-File with the suffix ``.py``.\n",
+    "\n",
+    "Import the classes you want to use from safeds.\n"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:24]:
```suggestion
    "Import the classes you want to use from Safe-DS.\n"
```

File: docs/tutorials/regression.ipynb (hunk 1 above)
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:11]:
```suggestion
    "This tutorial uses Safe-DS on **house sales data** to predict house prices.\n"
```

File: docs/tutorials/regression.ipynb (hunk 2 above)
Comment (lars-reimann) [reply docs/tutorials/regression.ipynb:24]: Which classes would that? Since later cells include the required imports, I'd rather remove this sentence here altogether.

File: docs/tutorials/regression.ipynb [hunk 3]
```diff
@@ -2,204 +2,391 @@
...
+   "metadata": {},
    "source": [
-    "from safeds.data.tabular.containers import Table\n",
+    "## Reading Data\n",
     "\n",
-    "pricing = Table.from_csv_file(\"data/house_sales.csv\")\n",
-    "# For visualisation purposes we only print out the first 15 rows.\n",
-    "pricing.slice_rows(0,15)"
-   ],
+    "Load your data into a `Table`, the data is available under `docs/tutorials/data/pricing.csv`:"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:33]: A clickable link would be nice.

File: docs/tutorials/regression.ipynb [hunk 4]
```diff
@@ -2,204 +2,391 @@
...
      "metadata": {},
      "output_type": "execute_result"
     }
+   ],
+   "source": [
+    "from safeds.data.tabular.containers import Table\n",
+    "\n",
+    "pricing = Table.from_csv_file(\"data/house_sales.csv\")\n",
+    "# For visualisation purposes we only print out the first 15 rows.\n",
+    "pricing.slice_rows(0,15)"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:89]:
```suggestion
    "pricing.slice_rows(length=15)"
```

Would also work and better describe the meaning of `15`.

File: docs/tutorials/regression.ipynb [hunk 5]
```diff
@@ -2,204 +2,391 @@
...
+    "pricing_outliers = pricing_values.remove_rows_with_outliers()\n",
+    "# For visualisation purposes we only print out the first 5 rows.\n",
+    "pricing_outliers.slice_rows(0,5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "See how to perform further Data-Cleaning at: https://library.safeds.com/en/stable/tutorials/data_processing"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:152]: Use relative links, so they also work locally/on the latest branch.

File: docs/tutorials/regression.ipynb [hunk 6]
```diff
@@ -2,204 +2,391 @@
...
    ],
+   "source": [
+    "# removes columns \"latitude\" and \"longitude\" from table\n",
+    "pricing_columns = pricing.remove_columns([\"latitude\", \"longitude\"])\n",
+    "# removes rows which contain missing values\n",
+    "pricing_values = pricing_columns.remove_rows_with_missing_values()\n",
+    "# removes rows which contain outliers\n",
+    "pricing_outliers = pricing_values.remove_rows_with_outliers()\n",
+    "# For visualisation purposes we only print out the first 5 rows.\n",
+    "pricing_outliers.slice_rows(0,5)"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:145]:
```suggestion
    "pricing_outliers.slice_rows(length=5)"
```

File: docs/tutorials/regression.ipynb [hunk 7]
```diff
@@ -2,204 +2,391 @@
...
+  {
+   "cell_type": "markdown",
    "metadata": {
     "collapsed": false
-   }
+   },
+   "source": [
+    "## Create Training and Testing Set\n",
+    "\n",
+    "Split the house sales dataset into two tables. A training set, that will be used later to implement a training model to predict the house prices. It contains 60% of the data. The testing set contains the rest of the data. Delete the column `price` from the test set, to be able to predict it later:\n"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:163]: Deleting the `price` column is no longer needed. An existing column simply gets overwritten during prediction.

File: docs/tutorials/regression.ipynb [hunk 8]
```diff
@@ -2,204 +2,391 @@
...
+    },
+    "collapsed": false
    },
-   "execution_count": 2,
-   "outputs": []
+   "outputs": [],
+   "source": [
+    "train_table, testing_table = pricing_outliers.split_rows(0.60)\n",
+    "\n",
+    "test_table = testing_table.remove_columns([\"price\"]).shuffle_rows()"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:180]: * `remove_columns` is not needed here.
* Why do we need to shuffle the test table?

File: docs/tutorials/regression.ipynb [hunk 9]
```diff
@@ -2,204 +2,391 @@
...
+     "output_type": "execute_result"
+    }
    ],
+   "source": [
+    "# removes columns \"latitude\" and \"longitude\" from table\n",
+    "pricing_columns = pricing.remove_columns([\"latitude\", \"longitude\"])\n",
+    "# removes rows which contain missing values\n",
+    "pricing_values = pricing_columns.remove_rows_with_missing_values()\n",
+    "# removes rows which contain outliers\n",
+    "pricing_outliers = pricing_values.remove_rows_with_outliers()\n",
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:143]: Could also chain the method calls, since we never refer to the intermediate results later.

File: docs/tutorials/regression.ipynb [hunk 10]
```diff
@@ -2,204 +2,391 @@
...
+    "collapsed": false
    },
-   "execution_count": 4,
-   "outputs": []
+   "outputs": [],
+   "source": [
+    "from safeds.ml.classical.regression import DecisionTreeRegressor\n",
+    "\n",
+    "model = DecisionTreeRegressor()\n",
+    "fitted_model = model.fit(train_tabular_dataset)"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:235]: Could be chained, since we don't need the untrained model later.

File: docs/tutorials/regression.ipynb [hunk 11]
```diff
@@ -2,204 +2,391 @@
...
      "metadata": {},
      "output_type": "execute_result"
     }
+   ],
+   "source": [
+    "prediction = fitted_model.predict(\n",
+    "    test_table\n",
+    ")\n",
+    "# For visualisation purposes we only print out the first 15 rows.\n",
+    "prediction.to_table().slice_rows(start=0, length=15)"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:303]:
```suggestion
    "prediction.to_table().slice_rows(length=15)"
```

For consistency, omit the start.

File: docs/tutorials/regression.ipynb [hunk 12]
```diff
@@ -2,204 +2,391 @@
...
+      ]
      },
-     "execution_count": 6,
+     "execution_count": 22,
      "metadata": {},
      "output_type": "execute_result"
     }
+   ],
+   "source": [
+    "test_tabular_dataset = testing_table.to_tabular_dataset(\"price\", extra_names=extra_names)\n",
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:340]: No need anymore to create a tabular dataset when using metrics. You can just pass the table directly.

File: docs/tutorials/regression.ipynb [hunk 13]
```diff
@@ -2,204 +2,391 @@
...
+    "\n",
+    "extra_names = [\"id\"]\n",
+    "train_tabular_dataset = train_table.to_tabular_dataset(\"price\", extra_names=extra_names)\n",
+    "\n",
+    "model = DecisionTreeRegressor()\n",
+    "fitted_model = model.fit(train_tabular_dataset)\n",
+    "prediction = fitted_model.predict(test_table)\n",
+    "\n",
+    "test_tabular_dataset = testing_table.to_tabular_dataset(\"price\", extra_names=extra_names)\n",
+    "fitted_model.mean_absolute_error(test_tabular_dataset)"
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:389]: Adjust accordingly.

File: docs/tutorials/regression.ipynb [hunk 14]
```diff
@@ -2,204 +2,391 @@
...
    "cell_type": "markdown",
-   "source": [
-    "4. Use `Decision Tree` regressor as a model for the regression. Pass the \"train_tabular_dataset\" table to the fit function of the model:\n"
-   ],
    "metadata": {
     "collapsed": false
-   }
+   },
+   "source": [
+    "## Creating and fitting a Regressor\n",
```
Comment (lars-reimann) [inline docs/tutorials/regression.ipynb:215]:
```suggestion
    "## Creating and Fitting a Regressor\n",
```

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Review by lars-reimann — APPROVED** [review]
A nice improvement, thank you.
